packit_service/worker/reporting.py

```diff
@@ -30,6 +69,22 @@ def __init__(
         self.commit_sha: str = commit_sha
         self.pr_id: Optional[int] = pr_id
 
+    @classmethod
+    def get(
+        cls, project: GitProject, commit_sha: str, pr_id: Optional[int] = None
+    ) -> "StatusReporter":
```

Comment:
Maybe some more specific name?

```suggestion
    def get_instance(
        cls, project: GitProject, commit_sha: str, pr_id: Optional[int] = None
    ) -> "StatusReporter":
```